src/Nethermind/Nethermind.Blockchain/Processing/BlockProcessor.cs

```diff
@@ -221,6 +223,7 @@ private TxReceipt[] ProcessTransactions(Block block, ProcessingOptions processin
                 StoreTxReceipts(block, receipts);
             }
 
+            _witnessCollector.Persist(block.Hash);
```

Comment:
Maybe persist should also reset before next block? That would remove that Reset call and method.
EDIT: Maybe resetting could be done when StateProvider is resseted?

Reply:
not sure, looked at it a bit, I feel it would be worse due to various exception / edge cases. my approach is very safe